### DISCUSSION TITLE
Pass a variable value from JS to SWF

### DISCUSSION THREAD

**Discussion by faniceto** [ruffle-rs/ruffle#1265] [Q&A]
Guys, we got a lot of e-learning courses that depend on loading some initial variables into swf to take the user to the correct class.
To do this we usually used JS to pass the variables in. Basically we find swf objects on page with getElementById and then call function SetVariable on that object.

Is it possible to do it with Ruffle? If so how can we do this?

**Comment by Justin-CB** [comment]
I'm not 100% sure about this, but I think that as long as you have the allowScriptAccess param on the object, it should work unless it's something different than what was recently implemented. If it still doesn't work, you can file an issue.

**Comment by n0samu** [comment]
If you are using the Flash plugin's built-in SetVariable function, then that is still not supported in Ruffle, see https://github.com/ruffle-rs/ruffle/issues/7868
